src/ocm/hooks/useClusterPreflightRequirements.ts

```diff
@@ -3,21 +3,24 @@ import { alertsSlice } from '../../common/reducers';
 import { Cluster, PreflightHardwareRequirements } from '../../common/api/types';
 import { ClustersAPI } from '../services/apis';
 import { getErrorMessage, handleApiError } from '../api';
+import { APIErrorMixin } from '../api/types';
+
+type ErrorType = APIErrorMixin & Error;
 
 const { addAlert } = alertsSlice.actions;
 
 const useClusterPreflightRequirements = (clusterId: Cluster['id']) => {
   const [preflightRequirements, setPreflightRequirements] =
     React.useState<PreflightHardwareRequirements>();
-  const [error, setError] = React.useState();
+  const [error, setError] = React.useState<ErrorType>();
```

Comment:
Not entirely related to your changes but I suspect that this should be a string - an error message rather than an ErrorType?

Reply:
I agree. Please look at the new commit https://github.com/openshift-assisted/assisted-ui-lib/pull/1456/commits/0e57cafb04cce22dd9b6866c5ebf23c87dde52bf

Changes done:
`useClusterPreflightRequirements`: reverted because it's addressed in your PR
`useInfraEnv`, `useInfraEnvId` and `useInfraEnvImageUrl`: these are not handled in your PR. They are now defined as strings given they are being used in reality more as booleans.